python/dgl/distributed/dist_graph.py

```diff
@@ -494,10 +495,20 @@ def rank(self):
         int
             The rank of the current graph store.
         '''
+        # If DistGraph doesn't have a local partition, it doesn't matter what rank
+        # it returns. There is no data locality any way, as long as the returned rank
+        # is unique in the system.
         if self._g is None:
-            return self._client.client_id
+            return rpc.get_rank()
         else:
-            return self._gpb.partid
+            # If DistGraph has a local partition, we should be careful about the rank
+            # we return. We need to return a rank that node_split or edge_split can split
+            # the workload with respect to data locality.
+            num_client = rpc.get_num_client()
+            num_client_per_part = num_client // self._gpb.num_partitions()
```

Comment:
Do we have the case that the number of clients can be different on each machine?

Reply:
If we will meet this case in the future, we can add a new API `get_num_client_per_machine(machine_id)`

Reply:
it sounds a very complex scenario. why do we want to run different numbers of clients on different machines?